cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go

```diff
@@ -56,3 +85,160 @@ func (s *Server) GetClient(ctx context.Context) (dynamic.Interface, error) {
+		AvailablePackagesSummaries: responsePackages,
+	}, nil
+}
+
+func AvailablePackageSummaryFromChart(chart *models.Chart) (*corev1.AvailablePackageSummary, error) {
+	pkg := &corev1.AvailablePackageSummary{}
+
+	if chart.Name == "" {
+		return nil, status.Errorf(codes.Internal, "required field .Name not found on helm package: %v", chart)
+	}
+	pkg.DisplayName = chart.Name
+
+	if chart.ChartVersions == nil || len(chart.ChartVersions) == 0 || chart.ChartVersions[0].Version == "" {
+		return nil, status.Errorf(codes.Internal, "required field .ChartVersions[0].Version not found on helm package: %v", chart)
+	}
+	pkg.LatestVersion = chart.ChartVersions[0].Version
+
+	if chart.Icon == "" {
+		return nil, status.Errorf(codes.Internal, "required field .Icon not found on helm package: %v", chart)
+	}
+	pkg.IconUrl = chart.Icon
+
+	if chart.Description == "" {
+		return nil, status.Errorf(codes.Internal, "required field .Description not found on helm package: %v", chart)
+	}
+	pkg.ShortDescription = chart.Description
+
+	if chart.ID == "" {
+		return nil, status.Errorf(codes.Internal, "required field .ID not found on helm package: %v", chart)
+	}
+	if chart.Repo == nil || chart.Repo.Namespace == "" {
+		return nil, status.Errorf(codes.Internal, "required field .Repo.Namespace not found on helm package: %v", chart)
+	}
+	pkg.AvailablePackageRef = &corev1.AvailablePackageReference{
+		Context:    &corev1.Context{Namespace: chart.Repo.Namespace},
+		Identifier: chart.ID,
+	}
+
+	return pkg, nil
+}
+
+func (s *Server) hasAccessToNamespace(ctx context.Context, namespace string) (bool, error) {
+	client, err := s.GetClient(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	// TODO(agamez): this is a temporary workaround since the client getter we (the plugin) get is just
+	// for dynamic interfaces. I guess we should also pass a normal k8s client in case the plugins want to
+	// interact with typed kubernetes resources.
+	selfSubjectAccessReviews := schema.GroupVersionResource{Group: "authorization.k8s.io", Version: "v1", Resource: "selfsubjectaccessreviews"}
+	unstructuredData, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&authorizationapi.SelfSubjectAccessReview{
+		Spec: authorizationapi.SelfSubjectAccessReviewSpec{
+			ResourceAttributes: &authorizationapi.ResourceAttributes{
+				Group:     "",
+				Resource:  "secrets",
+				Verb:      "get",
+				Namespace: namespace,
+			},
+		},
+	})
+	if err != nil {
+		return false, fmt.Errorf("Error parsing the selfSubjectAccessReviews request: %s", err)
+	}
+	selfSubjectAccessReviewsResponse, err := client.Resource(selfSubjectAccessReviews).Create(ctx, &unstructured.Unstructured{Object: unstructuredData}, metav1.CreateOptions{})
+	if err != nil {
+		return false, fmt.Errorf("Error creating the selfSubjectAccessReviews request: %s", err)
+	}
+	res := &authorizationapi.SelfSubjectAccessReview{}
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(selfSubjectAccessReviewsResponse.Object, res)
```

Comment:
This is what I meant by temporary ugly workaround :P . I just needed to use the dynamic client, so I converted it with `ToUnstructured` and with `FromUnstructured`

Reply:
Yep, see #3043 .